src/views/Zap/Zap.jsx

```diff
@@ -7,9 +7,14 @@ import ZapStakeAction from "./ZapStakeAction";
 import ZapInfo from "./ZapInfo";
 import { useAppSelector } from "src/hooks";
 import { Trans } from "@lingui/macro";
+import { usePathForNetwork } from "src/hooks/usePathForNetwork";
+import { useHistory } from "react-router";
 
 function Zap() {
   const { address, connect } = useWeb3Context();
+  const history = useHistory();
+  const networkId = useAppSelector(state => state.network.networkId);
+  usePathForNetwork({ pathName: "zap", networkID: networkId, history });
```

Comment:
this is how we can handle network based redirects in the least amount of lines I could imagine today. this rather than putting a useEffect on each file.